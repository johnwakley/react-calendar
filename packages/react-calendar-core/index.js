// Named exports
module.exports = {
  calendarYear: require('./lib/calendar').calendarYear,
  calendarMonth: require('./lib/calendar').calendarMonth,
  CalendarMonth: require('./lib/calendar').CalendarMonth,
  CalendarMonthEnum: require('./lib/calendar').CalendarMonthEnum,
  DEFAULT_LOCALE: require('./lib/calendar').DEFAULT_LOCALE,
  DatePresenter: require('./lib/DatePresenter').DatePresenter,
  DateStyle: require('./lib/DatePresenter').DateStyle,
  dateStyles: require('./lib/DatePresenter').dateStyles,
  MediaBreakpoints: require('./lib/media').MediaBreakpoints,
  mediaWithBreakpoints: require('./lib/media').mediaWithBreakpoints,
  media: require('./lib/media').media
}
