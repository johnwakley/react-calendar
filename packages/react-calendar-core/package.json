{
  "name": "react-calendar-core",
  "version": "0.1.0",
  "private": true,
  "dependencies": {
    "moment": "^2.18.1",
    "moment-range": "^3.0.3",
    "react": "^15.5.4",
    "react-dom": "^15.5.4",
    "styled-components": "^2.0.0"
  },
  "devDependencies": {
    "babel-cli": "^6.24.1",
    "babel-preset-es2015": "^6.24.1",
    "babel-preset-react-app": "^3.0.0",
    "flow-bin": "^0.47.0",
    "husky": "^0.13.3",
    "lint-staged": "^3.5.0",
    "prettier-standard": "^5.0.0",
    "react-scripts": "1.0.6",
    "standard": "^10.0.2"
  },
  "scripts": {
    "start": "react-scripts start",
    "build": "react-scripts build",
    "test": "yarn flow && standard && react-scripts test --env=jsdom",
    "test:dev": "react-scripts test --env=jsdom",
    "eject": "react-scripts eject",
    "flow": "flow",
    "format": "prettier-standard 'src/**/*.js'",
    "lint-staged": "lint-staged",
    "precommit": "CI=true yarn test",
    "prepush": "CI=true yarn test",
    "compile": "NODE_ENV=production babel src/ -d lib --presets=es2015,react-app"
  },
  "lint-staged": {
    "linters": {
      "src/**/*.js": [
        "prettier-standard",
        "git add"
      ]
    }
  },
  "standard": {
    "plugins": [
      "flowtype"
    ],
    "parser": "babel-eslint",
    "ignore": [
      "/lib/*"
    ],
    "globals": [
      "describe",
      "it",
      "expect",
      "beforeEach",
      "afterEach",
      "jest",
      "jasmine",
      "fit",
      "xit",
      "$Keys"
    ]
  }
}
