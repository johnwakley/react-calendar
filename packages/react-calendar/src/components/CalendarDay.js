/**
 * MIT License
 *
 * Copyright (c) 2017 johnwakley
 *
 * Permission is hereby granted, free of charge, to any person obtaining a copy
 * of this software and associated documentation files (the "Software"), to deal
 * in the Software without restriction, including without limitation the rights
 * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 * copies of the Software, and to permit persons to whom the Software is
 * furnished to do so, subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included in all
 * copies or substantial portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 * SOFTWARE.
 *
 */

// @flow

import React from 'react'
import type { Children } from 'react'
import styled from 'styled-components'
import { ical as theme } from '../themes'
import { dateStyles } from 'react-calendar-core'
import type { DateStyle } from 'react-calendar-core'

const markerColor = (dateStyle: DateStyle) => {
  if (dateStyle === dateStyles.TODAY) {
    return `
      color: ${theme.textColors.textInvert};
      background-color: ${theme.colors.red};
      border-radius: 1em;
    `
  }

  if (dateStyle === dateStyles.MUTED) {
    return `
      color: ${theme.textColors.textLighter};
    `
  }

  if (dateStyle === dateStyles.DAY_NAME) {
    return `
      color: ${theme.textColors.textLight};
    `
  }

  return `
    color: ${theme.textColors.text};
  `
}

const CalendarDay = ({
  children,
  dateStyle
}: {
  children?: Children,
  dateStyle: DateStyle
}) => {
  const StyledDay = styled.div`
    display: flex;
    flex: 1;
    justify-content: center;
    color: ${props =>
      props.muted
        ? props.theme.textColors.textLight
        : props.theme.textColors.text};
  `

  StyledDay.defaultProps = { theme }

  const StyledDayMarker = styled.div`
    text-align: center;
    width: 2em;
    height: 2em;
    line-height: 2em;
    white-space: nowrap;
    ${markerColor(dateStyle)}
  `

  return (
    <StyledDay>
      <StyledDayMarker>
        {children}
      </StyledDayMarker>
    </StyledDay>
  )
}

export default CalendarDay
