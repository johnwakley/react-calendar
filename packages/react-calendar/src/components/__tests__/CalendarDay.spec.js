import React from 'react'
import renderer from 'react-test-renderer'
import CalendarDay from '../CalendarDay'
import { dateStyles } from 'react-calendar-core'

// TODO: use jest-styled-components when stable

describe('CalendarDay', () => {
  it('should show today style', () => {
    const tree = renderer
      .create(<CalendarDay dateStyle={dateStyles.TODAY}>1</CalendarDay>)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('should show muted style', () => {
    const tree = renderer
      .create(<CalendarDay dateStyle={dateStyles.MUTED}>1</CalendarDay>)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('should show day_name style', () => {
    const tree = renderer
      .create(<CalendarDay dateStyle={dateStyles.DAY_NAME}>1</CalendarDay>)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('should show normal style', () => {
    const tree = renderer
      .create(<CalendarDay dateStyle={dateStyles.NORMAL}>1</CalendarDay>)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
