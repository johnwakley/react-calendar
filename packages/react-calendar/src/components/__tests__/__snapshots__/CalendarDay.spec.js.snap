// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CalendarDay should show day_name style 1`] = `
<div
  className="sc-ifAKCX fbhzVK"
>
  <div
    className="sc-EHOje drtFEP"
  >
    1
  </div>
</div>
`;

exports[`CalendarDay should show muted style 1`] = `
<div
  className="sc-htpNat jMMEso"
>
  <div
    className="sc-bxivhb jXNyRX"
  >
    1
  </div>
</div>
`;

exports[`CalendarDay should show normal style 1`] = `
<div
  className="sc-bZQynM cZCAer"
>
  <div
    className="sc-gzVnrw iyiVzV"
  >
    1
  </div>
</div>
`;

exports[`CalendarDay should show today style 1`] = `
<div
  className="sc-bdVaJa fTNhch"
>
  <div
    className="sc-bwzfXH kchIv"
  >
    1
  </div>
</div>
`;
